python/python_fastapi_server/routers/edr.py

```diff
@@ -0,0 +1,876 @@
+
+        parameters: dict(str, CovJsonParameter) = {}
+        ranges = {}
+
+        unit = CovJsonUnit(symbol="m3")
+        param = CovJsonParameter(
+            id=dat["name"],
+            observedProperty=CovJsonObservedProperty(label={"en": dat["name"]}),
+            unit=unit
+        )
+        #TODO: add units to CovJsonParameter
+        parameters[dat["name"]] = param
+        axis_names = ["x", "y", "t"]
+        shape = [1, 1, len(time_steps)]
+        if vertical_steps:
+            axis_names = ["x", "y", "z", "t"]
+            shape = [1, 1, len(vertical_steps), len(time_steps)]
+        _range = {
+            "axisNames": axis_names,
+            "shape": shape,
+            "values": values,
+        }
+        ranges[dat["name"]] = _range
+
+        axes: dict[str, ValuesAxis] = {
+            "x": ValuesAxis[float](values=[lon]),
+            "y": ValuesAxis[float](values=[lat]),
+            "t": ValuesAxis[AwareDatetime](values=[datetime.strptime(t, "%Y-%m-%dT%H:%M:%SZ").replace(tzinfo=timezone.utc) for t in time_steps]),
+        }
+        domain_type = "PointSeries"
+        if vertical_steps:
+            axes["z"] = ValuesAxis[float](values=vertical_steps)
+            if len(vertical_steps) > 1:
+                domain_type = "VerticalProfile"
+        if time_steps:
+            axes["t"] = ValuesAxis[AwareDatetime](values=[datetime.strptime(t, "%Y-%m-%dT%H:%M:%SZ").replace(tzinfo=timezone.utc) for t in time_steps])
+            if len(time_steps) > 1 and vertical_steps and len(
+                    vertical_steps) > 1:
+                domain_type = "Grid"
+
+        referencing = [
+            ReferenceSystemConnectionObject(
+                system=ReferenceSystem(
+                    type="GeographicCRS",
+                    id="http://www.opengis.net/def/crs/OGC/1.3/CRS84",
+                ),
+                coordinates=["x", "y"],
+            ),
+            ReferenceSystemConnectionObject(
+                system=ReferenceSystem(type="TemporalRS",
+                                       calendar="Gregorian"),
+                coordinates=["t"],
+            ),
+        ]
+        domain = Domain(domainType=domain_type,
+                        axes=axes,
+                        referencing=referencing)
+        covjson = Coverage(
+            id="test",
+            domain=domain,
+            ranges=ranges,
+            parameters=parameters,
+        )
+        if not fullcovjson:
+            fullcovjson = covjson
+        else:
+            for k, v in covjson.parameters.items():
+                fullcovjson.parameters[k] = v
+            for k, v in covjson.ranges.items():
+                fullcovjson.ranges[k] = v
+
+    return fullcovjson
+
+
+def set_edr_config():
+    config = {}
+    config["contact"] = {"email": "info@knmi.nl"}
+    config["provider"] = {"name": "KNMI", "url": "https://www.knmi.nl"}
+    config["keywords"] = ["HARM_N25", "precipitation"]
```

Comment:
This should probably come from the Edr configuration.
However at the moment we only have a EDR configuration settings per dataset and this is a server-wide configuration. There is no server-wide configuration mechanism at the moment.

I'll make this info even more general, so it will at least be correct.